### DISCUSSION TITLE
Possibility to compile version from current dev status?

### DISCUSSION THREAD

**Discussion by JohnDoeIsKIng** [nzymedefense/nzyme#576] [Questions] [answered]
Dear Lennart,
dear community,

is it possible to built a package (version, eventually as deb-file) from the current github source for test reasons?
Regarding a RasPi, on https://www.nzyme.org/download the latest version is 1.1.1, omitting the latest source modifications.
Any hints are appreciated!
Regards,

JD.

**Comment by lennartkoopmann** [comment, marked as answer]
You can check out the repository and run `mvn package`. This will generate a build `.jar` and `.deb` in `target/` and works on Linux as well as Windows systems.

> Regarding a RasPi, on https://www.nzyme.org/download the latest version is 1.1.1, omitting the latest source modifications.

Not sure what you mean with that. :)

**Reply by JohnDoeIsKIng** [reply]
Hey Lennart,

appreciated your reply. I did try your suggestion:
`[INFO] Running TestSuite
14:52:45.839 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:48.055 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:48.393 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:48.421 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:48.818 [main] INFO  org.reflections.Reflections - Reflections took 350 ms to scan 1 urls, producing 287 keys and 739 values 
14:52:48.992 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.439 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.469 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.492 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.519 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.545 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.569 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.590 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.614 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:49.635 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.101 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.126 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.153 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.171 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.192 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.211 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.229 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.246 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.268 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.282 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.303 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.327 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.346 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.366 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.385 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.399 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.413 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.430 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.448 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.463 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.476 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.491 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.511 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.522 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.538 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.574 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.586 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.598 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.609 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.620 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.632 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.645 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.658 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.681 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.699 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.715 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:50.733 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:51.440 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
14:52:51.460 [main] WARN  horse.wtf.nzyme.configuration.leader.LeaderConfigurationLoader - !!! DEPRECATION WANRING !!! The "graylog_uplinks" configuration has moved to a generic "uplinks" configuration. Please consult the configuration file reference on www.nzyme.org.
[ERROR] Tests run: 222, Failures: 52, Errors: 0, Skipped: 38, Time elapsed: 9.183 s <<< FAILURE! - in TestSuite
[ERROR] cleanEvents(horse.wtf.nzyme.events.EventServiceTest)  Time elapsed: 4.146 s  <<< FAILURE!
org.jdbi.v3.core.ConnectionException: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
	at horse.wtf.nzyme.events.EventServiceTest.cleanEvents(EventServiceTest.java:17)
Caused by: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
	at horse.wtf.nzyme.events.EventServiceTest.cleanEvents(EventServiceTest.java:17)
Caused by: java.net.ConnectException: Verbindungsaufbau abgelehnt (Connection refused)
	at horse.wtf.nzyme.events.EventServiceTest.cleanEvents(EventServiceTest.java:17)

[ERROR] cleanAlerts(horse.wtf.nzyme.alerts.AlertsServiceTest)  Time elapsed: 0.12 s  <<< FAILURE!
org.jdbi.v3.core.ConnectionException: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
	at horse.wtf.nzyme.alerts.AlertsServiceTest.cleanAlerts(AlertsServiceTest.java:25)
Caused by: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
	at horse.wtf.nzyme.alerts.AlertsServiceTest.cleanAlerts(AlertsServiceTest.java:25)
Caused by: java.net.ConnectException: Verbindungsaufbau abgelehnt (Connection refused)
	at horse.wtf.nzyme.alerts.AlertsServiceTest.cleanAlerts(AlertsServiceTest.java:25)

[ERROR] cleanDatabase(horse.wtf.nzyme.reporting.TacticalSummaryReportTest)  Time elapsed: 0.081 s  <<< FAILURE!
org.jdbi.v3.core.ConnectionException: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
	at horse.wtf.nzyme.reporting.TacticalSummaryReportTest.cleanDatabase(TacticalSummaryReportTest.java:25)
Caused by: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
	at horse.wtf.nzyme.reporting.TacticalSummaryReportTest.cleanDatabase(TacticalSummaryReportTest.java:25)
Caused by: java.net.ConnectException: Verbindungsaufbau abgelehnt (Connection refused)
	at horse.wtf.nzyme.reporting.TacticalSummaryReportTest.cleanDatabase(TacticalSummaryReportTest.java:25)
`

Do you have any hints what is causing these errors ? I don't know what to do with a "Connection to localhost:5432 refused."
Best regards

JD.

**Reply by lennartkoopmann** [reply]
It can't connect to PostgreSQL. Currently, you can't easily change (not permanently at least) the [test config files](https://github.com/lennartkoopmann/nzyme/tree/master/src/test/resources) so you'll have to configure a PostgreSQL like this:

Host: localhost
Port: 5432
Database: nzyme-java-tests
User: nzyme
No password

You can also build without running the tests:

```
$ mvn package -DskipTests
```

**Comment by JohnDoeIsKIng** [comment]
Thanks Lennart,

if I try to build the package on a Raspi3, it comes to a build error:

```
protoc-jar: downloading: https://repo.maven.apache.org/maven2/com/github/os72/protoc/3.11.4/protoc-3.11.4-linux-arm_32.exe
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  26.894 s
[INFO] Finished at: 2021-11-19T18:40:18+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.os72:protoc-jar-maven-plugin:3.11.4:run (default) on project nzyme: Error extracting protoc for version 3.11.4: Unsupported platform: protoc-3.11.4-linux-arm_32.exe -> [Help 1]
```

There seems to be no 32bit version of protoc for the ARM-architecture of the Raspi. I will try to compile it on myself and report further.

**Reply by lennartkoopmann** [reply]
(Edited for readability)

Yes, the `protoc` part will likely fail if you try to build it on ARM. It's designed to be compiled on `amd64` or `i368` systems running Windows or Linux. OSX will *probably* work, too.

**Comment by JohnDoeIsKIng** [comment]
Hi Lennart,

thanks for replying. So how do you build the Raspi-Deb in the Downloads-area ?
Best regards

JD.


Edit: On Ubuntu the current source build ends with
```
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ nzyme ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 437 source files to /home/john/Githubs/nzyme/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[7,71] cannot find symbol
  symbol: class ContactRecorderFrameCountHistogramEntry
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[61,77] package ContactRecorderFrameCountHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[67,51] package ContactRecorderFrameCountHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[72,51] package ContactRecorderFrameCountHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[80,12] cannot find symbol
  symbol:   class ContactRecorderFrameCountHistogramEntry
  location: class horse.wtf.nzyme.bandits.engine.AutoValue_ContactRecorderFrameCountHistogramEntry.Builder
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[7,75] cannot find symbol
  symbol: class ContactRecorderSignalStrengthHistogramEntry
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[61,81] package ContactRecorderSignalStrengthHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[67,55] package ContactRecorderSignalStrengthHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[72,55] package ContactRecorderSignalStrengthHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[80,12] cannot find symbol
  symbol:   class ContactRecorderSignalStrengthHistogramEntry
  location: class horse.wtf.nzyme.bandits.engine.AutoValue_ContactRecorderSignalStrengthHistogramEntry.Builder
[INFO] 10 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:31 min
[INFO] Finished at: 2021-11-21T13:00:26+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project nzyme: Compilation failure: Compilation failure: 
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[7,71] cannot find symbol
[ERROR]   symbol: class ContactRecorderFrameCountHistogramEntry
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[61,77] package ContactRecorderFrameCountHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[67,51] package ContactRecorderFrameCountHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[72,51] package ContactRecorderFrameCountHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderFrameCountHistogramEntry.java:[80,12] cannot find symbol
[ERROR]   symbol:   class ContactRecorderFrameCountHistogramEntry
[ERROR]   location: class horse.wtf.nzyme.bandits.engine.AutoValue_ContactRecorderFrameCountHistogramEntry.Builder
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[7,75] cannot find symbol
[ERROR]   symbol: class ContactRecorderSignalStrengthHistogramEntry
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[61,81] package ContactRecorderSignalStrengthHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[67,55] package ContactRecorderSignalStrengthHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[72,55] package ContactRecorderSignalStrengthHistogramEntry does not exist
[ERROR] /home/john/Githubs/nzyme/target/generated-sources/annotations/horse/wtf/nzyme/bandits/engine/AutoValue_ContactRecorderSignalStrengthHistogramEntry.java:[80,12] cannot find symbol
[ERROR]   symbol:   class ContactRecorderSignalStrengthHistogramEntry
[ERROR]   location: class horse.wtf.nzyme.bandits.engine.AutoValue_ContactRecorderSignalStrengthHistogramEntry.Builder
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

**Reply by lennartkoopmann** [reply]
Use three ` ``` ` to format whole blocks of code or output.

It builds for me:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:43 min
[INFO] Finished at: 2021-11-21T09:58:03-06:00
[INFO] ------------------------------------------------------------------------
```

Try a `mvn clean` first and make sure you pulled the latest version from git.

**Comment by JohnDoeIsKIng** [comment]
Hi Lennart,

thanks for the hints.
I've built it successfully with -DskipTests on an ubuntu system an installed it on a raspi 3.
I'll gamble around a bit with it the next days. ;-)
Best regards

JD.

**Reply by lennartkoopmann** [reply]
Sounds great!
